src/Libraries/Microsoft.Extensions.Telemetry.Abstractions/README.md

```diff
@@ -18,6 +23,100 @@ Or directly in the C# project file:
 </ItemGroup>
 ```
 
+## Usage
+
+### Enhanced Logging Capabilities
+
+The package includes a custom logging generator that enhances the default .NET logging capabilities by replacing the default generator. This generator automatically logs the contents of collections and offers advanced logging features, significantly improving the debugging and monitoring process.
+
+```csharp
+[LoggerMessage(1, LogLevel.Information, "These are the contents of my dictionary: {dictionary}")]
+internal static partial void LogMyDictionary(ILogger<Program> logger, Dictionary<int, string> temperature);
+```
+
+It also adds the `LogProperties` attribute which can be applied to a an object parameter of a `LoggerMessage` method. It introspects the passed-in object and automatically adds tags for all its properties. This leads to more informative logs without the need for manual tagging of each property.
+
+```csharp
+[LoggerMessage(1, LogLevel.Information, "Detected a new temperature: {temperature}")]
+internal static partial void LogNewTemperature(ILogger<Program> logger, [LogProperties] Temperature temperature);
+
+internal record Temperature(double value, TemperatureUnit unit);
+```
+
+### Log Enrichment
+
+To enrich your logging data, you can add custom log enrichers to your service collection. This can be done using specific implementations or generic types.
```

Comment:
```suggestion
Logging data can be enriched by adding custom log enrichers to the service collection. This can be done using specific implementations or generic types.
```